apps/src/aichat/chatApi.ts

```diff
@@ -0,0 +1,71 @@
+import {ChatCompletionMessage} from './types';
+import HttpClient from '@cdo/apps/util/HttpClient';
+import {Role, Status} from './constants';
+
+export async function postOpenaiChatCompletion(
+  chatMessages: ChatCompletionMessage[]
+): Promise<ChatCompletionResponse> {
+  const payload = {messages: formatForChatCompletion(chatMessages)};
+  // Send request to chat completion backend controller.
+  const url = '/openai/chat_completion';
+  const response = await HttpClient.post(url, JSON.stringify(payload), true, {
+    'Content-Type': 'application/json; charset=UTF-8',
+  });
+  if (response.status === 200) {
+    return await response.json();
+  } else {
+    return null;
+  }
+}
+
+const formatForChatCompletion = (chatMessages: ChatCompletionMessage[]) => {
+  return chatMessages.map(message => {
+    return {role: message.role, content: message.chatMessageText};
+  });
+};
+
+export async function getChatCompletionMessage(
+  systemPrompt: string,
+  lastMessageId: number,
+  newMessage: string,
+  chatMessages: ChatCompletionMessage[]
+): Promise<ChatCompletionMessage[]> {
```

Comment:
Might be good to be more specific about the response type here. What about something like
```
{
   status: 'ok' | 'inappropriate' | ...etc
   assistantResponse?: string
}
```
With the understanding that if status is not `ok`, then `assistantResponse` will be undefined. This way, this API component doesn't need to know too much about how to form a `ChatCompletionMessage` and managing additional fields like id, name, etc.

Reply:
Sounds good - thank you!